### ISSUE TITLE
CTC loss is 0 on GPU Tesla V100

### ISSUE THREAD

**Issue by bjlgcxc** [SeanNaren/warp-ctc#162]
I get the problem of continuously get zero loss (loss = 0) on GPU Tesla V100，if anyone got the same problem or have some solutions?

**Comment by raotnameh** [comment]
facing the same problem on longer audios of around 40-60 seconds.
@SeanNaren

**Comment by bjlgcxc** [comment]
I have solved the problem, just put the input of ctc loss on cpu 

for example ,  CTC(acts, labels, act_lens, label_lens), 
  
   do acts.cpu()  before run CTC loss

**Comment by Stonesjtu** [comment]
The ctc loss will return 0 on errors. One is the ctc loss is not compiled with cuda support but got cuda tensor input. Another situation is input sequence is too long.

**Comment by IAASSIBLCU** [comment]
> The ctc loss will return 0 on errors. One is the ctc loss is not compiled with cuda support but got cuda tensor input. Another situation is input sequence is too long.

I also face the same problem. so I want to know how to check the ctc loss is compiled with cuda or not. and, if it really isn't compiled, what can I do to make them compiled. thanks a lot!!!

**Comment by raotnameh** [comment]
@IAASSIBLCU I would recommend you to try using PyTorch CTC.

**Comment by SeanNaren** [comment]
I too would also suggest using PyTorch CTC :) this library is fairly old, and I've also moved this project to use it successfully: https://github.com/SeanNaren/deepspeech.pytorch

**Comment by IAASSIBLCU** [comment]
@raotnameh  @SeanNaren  thanks a lot!! I will try PyTorch CTC in my project. Hope to get the dream result.
